torch/utils/_pytree.py

```diff
@@ -599,6 +607,8 @@ def _treespec_to_json(spec: TreeSpec) -> _TreeSpecSchema:
     if spec.type not in SUPPORTED_SERIALIZED_TYPES:
         raise NotImplementedError(
             f"Serializing {spec.type} in pytree is not registered."
+            "You can register using torch.utils._pytree._register_pytree_node "
```

Comment:
This error message is correct, but nevertheless I can imagine this being a little confusing to users ("but I *did* register my pytree!!"), who may not parse all this business with fqns as being the crucial element.

To make this more instantly fixable by users, I suggest: if `type_fqn is None`  on L142 above, then instead of skipping adding to `SUPPORTED_SERIALIZED_TYPES`, we serialize a dummy `NoTypeFqnFound` type. Then, in the error message here, we can check for the presence of that type, and offer a much more direct error message, e.g. "please update your _register_pytree_node call with a `type_fqn` kwarg".

I think this takes the percentage of cases that will need direct support from us from 20% to like 1%, which is nice.